kin-recovery/recovery-sample/src/main/java/recovery/sample/BackupAndRestoreActivity.java

```diff
@@ -65,73 +69,110 @@ public void onClick(View v) {
 		if (vId == R.id.backup_current_account) {
 			backupAndRestorePresenter.backupClicked();
 		} else if (vId == R.id.recover_account) {
+			setBalanceProgressBar(true);
 			backupAndRestorePresenter.restoreClicked();
 		} else if (vId == R.id.create_new_account) {
+			setBalanceProgressBar(true);
+			setPublicAddressProgressBar(true);
 			createNewAccount.setEnabled(false);
+			backupCurrentAccount.setEnabled(false);
+			recoverAccount.setEnabled(false);
 			backupAndRestorePresenter.createAccountClicked();
 		}
 	}
 
+	private void setBalanceProgressBar(boolean show) {
+		balanceProgress.setVisibility(show ? View.VISIBLE : View.GONE);
+	}
+
+	private void setPublicAddressProgressBar(boolean show) {
+		publicAddressProgressBar.setVisibility(show ? View.VISIBLE : View.GONE);
+	}
+
 	@Override
 	protected void onActivityResult(int requestCode, int resultCode, Intent data) {
 		super.onActivityResult(requestCode, resultCode, data);
-		backupAndRestorePresenter.onActivityResult(requestCode, resultCode, data);
+		if (requestCode == REQ_CODE_BACKUP || requestCode == REQ_CODE_RESTORE) {
```

Comment:
Yossi Segev want it to be controlled by the developer so i used it in the sample app.
Meaning that the developer can check what was the request code because he may have many options for results.
Regarding the clear accounts - I will remove the clear account code.